### ISSUE TITLE
Widget - not always updating time.

### ISSUE THREAD

**Issue by NitroNils** [SimpleMobileTools/Simple-Clock#62]
Sometimes I notice that the clock widget hangs/stops counting time.

I don't know _what_  causes it or _when_ it stops. Suddenly it's back online again.
Happens fairly regularly.

Upon clicking the widget the clock app is fully synced, even though the widget isn't.

Probably not your fault, because my Huawei P8 Lite has some strange issues with permissions or smth I suspect.

Do you have any clues?

**Comment by tibbi** [comment]
Well, widgets in general are quite buggy on Android. If I break something, it can block widgets of all apps from refreshing themselves, which is just ridiculous. Just a device reboot fixes it. Im always trying to improve it but it isnt that simple.

**Comment by NitroNils** [comment]
@tibbi So a tough one.
Thanks for clearing up a bit.

**Comment by Joppla** [comment]
@tibbi: I like the idea of open source software very much. But was is the sense of a clock widget without the correct time? I have now - as test - 3 clock widget installed on my phone and the original. 3 working correct, only tours not. Why don't you look f.e. in the open source clock widget of FairPhone how they do it? That widget works correct. Every time rebooting the phone seems me not the correct solution.

**Comment by tibbi** [comment]
It works just fine to me. Such issues are usually device specific and if I cant reproduce something, it is hard to fix.

**Comment by Joppla** [comment]
Ok, in that case I have to remove your app.

Met vriendelijke groet,

Jop

Op 29-3-2020 om 21:33 schreef Tibor Kaputa:

**Comment by Brane1003** [comment]
What worked for me on my Huawei device... First delete data and uninstall default clock app and then install Simple Clock. Hope it helps :)

**Comment by inson1** [comment]
@Joppla @NitroNils Its few years old issue. Did you try new versions? Did you try it with new phone?

**Comment by Joppla** [comment]
> @Joppla @NitroNils Its few years old issue. Did you try new versions? Did you try it with new phone?
> 
> Or https://dontkillmyapp.com/

No, why should I? The developer wasn't interested in researching the problem. See: 
> It works just fine to me. Such issues are usually device specific and if I cant reproduce something, it is hard to fix.

**Comment by NitroNils** [comment]
@inson1 Indeed an old issue. No easy way for me to test as I don't use Android as a daily driver anymore.

@Joppla  That's unfair. He never claimed he was uninterested in fixing it, but that it was hard to reproduce the error (without the same device as you have).

Hopefully Google has improved the widget architecture since those days. They are to blame, not @tibbi!

Be grateful that he has shown such dedication over so many years. A matter of respect. You are not even entitled to get any help or support whatsoever (according to the license).
